### PR TITLE
add ability to allocate to vaults

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
operator and governance have the ability to allocate funds from main vault to sub vaults

### Test Plan
- unit tests
- localnet testing

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced the `allocateToVault` method for fund allocation between vaults.
  - Added new request and response types for vault allocation, enhancing control over fund management.
  - Implemented a new RPC method `AllocateToVault` to facilitate fund transfers.
  - Added a new command in the CLI for managing vault allocations.

- **Bug Fixes**
  - Improved error handling for invalid authority in vault management functions.

- **Tests**
  - Added comprehensive unit tests for the `AllocateToVault` functionality, covering various allocation scenarios.

- **Documentation**
  - Updated error messages for consistency in authority validation tests.
  - Enhanced the `sample_pregenesis.json` with a defined subaccount structure.

### CODE REVIEW

File: protocol/x/vault/keeper/msg_server_allocate_to_vault.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	assetstypes "github.com/dydxprotocol/v4-chain/protocol/x/assets/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// AllocateToVault allocates funds from main vault to a vault.
+func (k msgServer) AllocateToVault(
```
Comment (vincentwschau) [inline protocol/x/vault/keeper/msg_server_allocate_to_vault.go:14]: What happens if the vault doesn't exist in state?

Comment (tqin7) [reply protocol/x/vault/keeper/msg_server_allocate_to_vault.go:14]: i was thinking that the operator needs to separately set vault params to have a vault exist in state

a smoother experience is probably to set vault params with `stand_by` status here during allocation if the allocated-to vault doesn't exist

Comment (vincentwschau) [reply protocol/x/vault/keeper/msg_server_allocate_to_vault.go:14]: >i was thinking that the operator needs to separately set vault params to have a vault exist in state

Is this validated anywhere? What happens if the operator doesn't make the vault exist?

Comment (tqin7) [reply protocol/x/vault/keeper/msg_server_allocate_to_vault.go:14]: not validated so I'd vote to set vault params with `stand_by` status during allocation. wdyt?

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**